### DISCUSSION TITLE
Can't use import variables inside getStaticProps or getServerSideProps

### DISCUSSION THREAD

**Discussion by schienieder** [vercel/next.js#26966] [Help]
Hello I am new to nextjs and I am using it with django drf and postgres. The problem is I can't use 'js-cookie' in which I imported on top of my profile.js file inside getStaticProps neither getServerSideProps like this:

My import:
`import Cookies from 'js-cookie'
import jwt_decode from 'jwt-decode'`

My use case inside getStaticProps and getServerSideProps. And when I console.log(token) it returns undefined:
`export const getStaticProps = async () => {
    const token = Cookies.get('jwt')
    console.log(token)
    const decoded_token = jwt_decode(token)
    const res = await fetch(`http://localhost:8000/profile/${decoded_token.user_id}`, {
        method : 'GET',
        headers : {'Authorization' : 'Bearer'+' '+token}
    })
    const data = await res.json()
    return {
        props : {
            clientProfile : data,
        }
    }

export const getServerSideProps = async () => {
    const token = Cookies.get('jwt')
    console.log(token)
    const decoded_token = jwt_decode(token)
    const res = await fetch(`http://localhost:8000/profile/${decoded_token.user_id}`, {
        method : 'GET',
        headers : {'Authorization' : 'Bearer'+' '+token}
    })
    const data = await res.json()
    return {
        props : {
            clientProfile : data,
        }
    }
}`

My use case not in getStaticProps or getServerSideProps but it works and it reads the token correctly:
`export default function profile({ clientProfile }) {
const router = useRouter()
    const axios = require('axios')
    const readCookie = () => {
        try {
            const jwt_token = Cookies.get('jwt')
            const decoded_token = jwt_decode(jwt_token)
            axios({
                method : 'GET',
                url : `http://localhost:8000/account/${decoded_token.user_id}`,
                headers : {'Authorization' : 'Bearer'+' '+ jwt_token}
            })
            .then((response) => {
                response.data.role !== 'client' ? router.push('/login') : ''
                axios({
                    method : 'GET',
                    url : `http://localhost:8000/profile/${decoded_token.user_id}`,
                    headers : {'Authorization' : 'Bearer'+' '+jwt_token}
                })
                .then((response) => {
                    console.log(response.data)
                })
                .catch((error) => {
                    console.log(error.response)
                })
            })
            .catch((error) => {
                Swal.fire({
                    icon : 'error',
                    title: 'Error',
                    text: `${error.response}`,
                    showCloseButton: true,
                    confirmButtonColor: '#0F766E',
                })
                console.log(error.response)
            })
            console.log(jwt_token)
        }
        catch {
            router.push('/login')
        }
    }
    useEffect(() => {
        readCookie()
    }, [])
}`

In addition, it also works if I directly paste the jwt but it is not dynamic and I don't want that like so:

`export const getStaticProps = async () => {
    const token = 'my_jwt_example_like_so'
    const res = await fetch('http://localhost:8000/profile/1', {
        method : 'GET',
        headers : {'Authorization' : 'Bearer'+' '+token}
    })
    const data = await res.json()
    return {
        props : {
            clientProfile : data,
        }
    }
}`

**Comment by wwwdepot** [comment]
`getStaticProps` and `getServerSideProps` are server-side functions, using a browser library is not gonna work. To read cookie in node.js environment, you might want to try the following:

```js
import cookie from "cookie";

export const getServerSideProps = async ({ req }) => {
  const cookies = cookie.parse(req.headers.cookie || '');
  const decoded_token = jwt_decode(cookies.jwt);

  ...
};
```

**Comment by schienieder** [comment]
Apparantly solved this one, all I need to do was pass req in like so:
getServerSideProps(req)

And get the cookie using req.cookies.jwt, now it works
